### PR TITLE
Isolation Forest

### PR DESCRIPTION
**Pull request by oualib**
- Adding all the methods for Isolation Forest
 - Indenting the code

@afard can you please review the code? I am not sure for the scoring method.

@ansleis can you please review the new descriptions in the files: ensemble.py, memmodel.py

TODO:
Adding the tests when the 12.x RPM is available

### CODE REVIEW

**Review by ansleis — APPROVED** [review]

File: verticapy/learn/ensemble.py [hunk 1]
```diff
@@ -416,6 +416,54 @@ def get_prior(self):
...
+    """
+---------------------------------------------------------------------------
+Creates an IsolationForest object using the Vertica IFOREST algorithm.
+
+Parameters
+----------
+name: str
+    Name of the the model. The model is stored in the DB.
+n_estimators: int, optional
+    The number of trees in the forest, an integer between 0 and 1000, inclusive.
```
Comment (afard) [inline verticapy/learn/ensemble.py:430]: Only the upper bound is inclusive.
Valid range: (0, 1000]

File: verticapy/learn/ensemble.py [hunk 2]
```diff
@@ -416,6 +416,54 @@ def get_prior(self):
...
+max_depth: int, optional
+    Maximum depth of each tree.
+nbins: int, optional
```
Comment (afard) [inline verticapy/learn/ensemble.py:433]: I wonder if you want to mention about the valid range of max_depth and nbins as well.

**Review by afard — APPROVED** [review]
I've left a few minor comments.

File: verticapy/learn/ensemble.py (hunk 1 above)
Comment (oualib) [reply verticapy/learn/ensemble.py:430]: Corrected

File: verticapy/learn/ensemble.py (hunk 2 above)
Comment (oualib) [reply verticapy/learn/ensemble.py:433]: Done;

**Pull request merged**